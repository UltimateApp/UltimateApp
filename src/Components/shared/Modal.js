import React, { useState, useEffect } from 'react';
import { StyleSheet, Modal as NativeModal, View, Text, TouchableOpacity } from 'react-native';

import theme from '../../styles/theme.style';
import I18n from '../../utils/i18n';

const Modal = ({ title, visible, onClose, children }) => {
  return (
    <NativeModal animationType="fade" transparent useNativeDriver visible={visible} onRequestClose={onClose}>
      <TouchableOpacity style={styles.backdrop} activeOpacity={1} onPress={onClose} />
      <View style={styles.centeredView}>
        <View style={styles.modalView}>
          <Text style={styles.titleText}>{title}</Text>
          {children}
          <TouchableOpacity style={styles.returnButton} onPress={onClose}>
            <Text style={styles.returnButtonText}>{I18n.t('shared.back')}</Text>
          </TouchableOpacity>
        </View>
      </View>
    </NativeModal>
  );
};

const styles = StyleSheet.create({
  backdrop: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    backgroundColor: 'rgba(0,0,0,0.5)',
  },
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  modalView: {
    maxWidth: '80%',
    backgroundColor: 'white',
    borderRadius: 20,
    padding: 20,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 10,
  },
  titleText: {
    fontWeight: 'bold',
    fontSize: 20,
    marginBottom: 20,
    textAlign: 'center',
  },
  returnButton: {
    borderColor: theme.MAIN_COLOR,
    borderRadius: 10,
    borderWidth: 1,
    marginTop: 20,
    paddingVertical: 10,
    paddingHorizontal: 20,
    width: 120,
  },
  returnButtonText: {
    color: theme.MAIN_COLOR,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});

export default Modal;
