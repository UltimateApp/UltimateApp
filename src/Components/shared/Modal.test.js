import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

import Modal from './Modal';

beforeEach(() => jest.useFakeTimers()); // for Modal behaviour

describe('<Modal />', () => {
  it('renders correctly', () => {
    const onClose = jest.fn();
    const tree = renderer
      .create(
        <Modal visible title="My Title" onClose={onClose}>
          <Text>Hello</Text>
        </Modal>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
