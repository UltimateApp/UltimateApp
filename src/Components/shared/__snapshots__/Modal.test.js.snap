// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  useNativeDriver={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "bottom": 0,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "elevation": 10,
          "maxWidth": "80%",
          "padding": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        My Title
      </Text>
      <Text>
        Hello
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#2a9e91",
            "borderRadius": 10,
            "borderWidth": 1,
            "marginTop": 20,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2a9e91",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Back
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
